core/src/test/scala/org/apache/spark/status/AppStatusListenerSuite.scala

```diff
@@ -1930,6 +1930,78 @@ abstract class AppStatusListenerSuite extends SparkFunSuite with BeforeAndAfter
     assert(job.numActiveStages == 0)
   }
 
+  test("SPARK-50356: skip shared stage for all jobs") {
+    val listener = new AppStatusListener(store, conf, live = true)
+
+    // This will be a shared stage in PENDING state which gets SKIPPED status
+    // when first job is completed
+    val sharedStageId = 1
+    val sharedPendingStage = new StageInfo(sharedStageId, 0, "sharedStage", 10,
```

Comment:
NIT: I would introduce a `val` named as `sharedStageAttemptId` as that 0 (especially down in the Array is bit cryptic for me).

Reply:
Thanks for pointing out now I'm using the new value.